tests/LazyRead.hs

```diff
@@ -1,12 +1,10 @@
--- Ensure that readInt and readInteger over lazy ByteStrings are not
--- excessively strict.
-
-module Main (main) where
-
 import Data.ByteString.Char8 (pack)
 import Data.ByteString.Lazy.Char8 (readInt, readInteger)
 import Data.ByteString.Lazy.Internal (ByteString(..))
 
+-- Ensure that readInt and readInteger over lazy ByteStrings are not
```

Comment:
I've put this in but feels like it should be converted to do some sort of an equality tests?

Reply:
The crucial point here is that `readInt` / `readInteger` of `Chunk (pack "2z") undefined` does not evaluate `undefined`. But yes, bonus points for checking that `readInt` / `readInteger` actually returns `1` or `2` :)

Reply:
cool will look at that, because there is something close which already exists but not quite what we're actually after. 

https://github.com/haskell/bytestring/blob/aca709af754a8a19a317004175d6c71224ce4094/tests/Properties.hs#L1319

We're testing the lazyness of `readInt` / `readInteger` :)